### ISSUE TITLE
Request to make a temporary doc change to satisfy the IntelliJ-Rust parser

### ISSUE THREAD

**Issue by SaadiSave** [tokio-rs/axum#882]
Is this bit of code

![image](https://user-images.githubusercontent.com/53120037/159684049-69e8ce9f-6c4e-4230-8472-b968094d81bc.png)

absolutely necessary? If not, intellij-rust/intellij-rust#8589 can be worked around by removing this line (``#[doc = concat!("Route `", stringify!($method) ,"` requests to the given handler.")]``) from all the `*_handler_fn` macros, until intellij-rust is fixed. Please consider doing so.

**Comment by jplatte** [comment]
I don't think it's reasonable to remove important parts of the docs to work around an IDE issue. I just tried one way of simplifying the implementation that might have fixed it if it worked, but it didn't work.

**Comment by davidpdrsn** [comment]
Yep I agree. I don't wanna remove docs due to IDE bugs.

You're welcome to suggest alternate workarounds.

**Comment by jplatte** [comment]
For the record, what I tried is moving the docs out of the inner macro invocation. This didn't work for two reasons:

* Apparently putting docs on macro invocations that produce items to document those items doesn't work
* `#[doc = concat!()]` seems to somehow need the extra macro layer to function (I observed this in a different project and found on StackOverflow or such that people tend to use a `doc_concat!` macro for this)

**Comment by SaadiSave** [comment]
> Yep I agree. I don't wanna remove docs due to IDE bugs.
> 
> You're welcome to suggest alternate workarounds.

That's okay. This is not high priority.

> For the record, what I tried is moving the docs out of the inner macro invocation. This didn't work for two reasons:
> 
> * Apparently putting docs on macro invocations that produce items to document those items doesn't work
> * `#[doc = concat!()]` seems to somehow need the extra macro layer to function (I observed this in a different project and found on StackOverflow or such that people tend to use a `doc_concat!` macro for this)

Is the second option possible? I know that adding a new dependency for something so trivial isn't ideal, but would it be acceptable?

**Comment by davidpdrsn** [comment]
I'm not aware of what `doc_concat!` is. Couldn't find a crate with that name.

**Comment by jplatte** [comment]
It's not a crate, just a macro that's being copy-pasted around. I'm pretty sure it's going to break IntelliJ in the same way. Here's the one I've added to [Ruma](https://github.com/ruma/ruma) a while ago:

```rust
macro_rules! doc_concat {
    ( $( #[doc = $doc:expr] $( $thing:tt )* )* ) => ( $( #[doc = $doc] $( $thing )* )* );
}
```

The issue I referenced in a comment above it, https://github.com/rust-lang/rust/issues/52607, was apparently closed by https://github.com/rust-lang/rust/pull/78837 but I'm still seeing issues without the macro, not sure what to make of that. There's a small chance it was just my own editor integration (rust-analyzer) that broke, but regardless of that I don't see how we could fix the first issue I mentioned.
